source/slang/slang-diagnostics.cpp

```diff
@@ -177,6 +177,42 @@ static void formatDiagnostic(
             humaneLoc = sourceView->getHumaneLoc(sourceLoc);
         }
         formatDiagnostic(humaneLoc, diagnostic, sb);
+
+        {
+            SourceView* currentView = sourceView;
+
+            while (currentView && currentView->getInitiatingSourceLoc().isValid() && currentView->getSourceFile()->getPathInfo().type == PathInfo::Type::TokenPaste)
```

Comment:
To make sure I understand; this is specifically only working up the stack of source views in the case where the error location is inside the results of a token paste?

If that is correct, then this seems like a good way to introduce this functionality without making large-scale changes to our diagnostic output in all cases.

Eventually we will likely want to turn on the printing of the full "backtrace" on more error messages, but it seems like there are open questions about what the policy for that should be (every message? every non-note message?)

Reply:
Yes - it's only working up 'stack' from pastes. That I suspect from there we want to do the backtrace of macro applications, as is touched on in issue #1600. You might want also to back trace includes - and I would agree there are open issues on how this should be done, where, and how to control it.